src/types/bytes.rs

```diff
@@ -97,6 +97,24 @@ impl PyBytes {
     }
 }
 
+impl Py<PyBytes> {
+    /// Gets the Python bytes as a byte slice. Because Python bytes are
+    /// immutable, the result may be used for as long as the reference to
+    /// `self` is held.
+    pub fn as_bytes<'a>(&self, _py: Python<'_>) -> &'a [u8] {
```

Comment:
This produces a slice with an unbounded lifetime. It must be bound on the lifetime of self.

```suggestion
    /// Gets the Python bytes as a byte slice. Because Python bytes are
    /// immutable, the result may be used for as long as the reference to
    /// `self` is held.
    pub fn as_bytes<'a>(&'a self, _py: Python<'_>) -> &'a [u8] {
```

Reply:
As far as I understand it is OK to use this slice after dropping the GIL (?). This is probably a good idea to clarify in the docs.

Reply:
a) Whoops, that was an accident, I'd meant to put `&'a self`,
b) docs added